heron/packing/src/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPacking.java

```diff
@@ -153,7 +163,45 @@ public PackingPlan pack() {
     Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
 
     Set<PackingPlan.ContainerPlan> containerPlans = PackingUtils.buildContainerPlans(
-        containerInstances, ramMap, this.defaultInstanceResources, paddingPercentage);
+        resourceCompliantRRAllocation.get(), ramMap, this.defaultInstanceResources,
+        paddingPercentage);
+    return new PackingPlan(topology.getId(), containerPlans);
+  }
+
+  /**
+   * Get a new packing plan given an existing packing plan and component-level changes.
+   *
+   * @return new packing plan
+   */
+  public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges) {
+    int adjustments = 0;
+    this.numAdjustments = 0;
+    this.numContainers = currentPackingPlan.getContainers().size();
+    this.containerId = 1;
+
+    int additionalContainers = computeNumAdditionalContainers(componentChanges, currentPackingPlan);
+    increaseNumContainers(additionalContainers);
+    LOG.info(String.format("Allocated "
+            + "%s additional containers for repack bring the number of containers to %s.",
+        additionalContainers, this.numContainers));
+    // Get the instances using Resource Compliant Round Robin allocation
+    Optional<Map<Integer, List<InstanceId>>> resourceCompliantRRAllocation =
+        getResourceCompliantRRAllocation(currentPackingPlan, componentChanges);
+
+    while (!resourceCompliantRRAllocation.isPresent()) {
```

Comment:
can you explain in comments what's happening here? it seems like we just keep calling `getResourceCompliantRRAllocation` until it returns something, but nothing changes in the values we pass to it so it's unclear why we should expect a different result each time.